### PR TITLE
fix(checkout): DATA-6891 missing transactions

### PR DESCRIPTION
**Pull request by bc-maxym-parfenov**
## What?
[DATA-6891](https://jira.bigcommerce.com/browse/DATA-6891)

This PR is trying to fix an issue with missing transactions and products in GA

## Testing / Proof
Screenshot 2020-08-21 at 22 20 34
Screenshot 2020-08-21 at 22 20 25
Screenshot 2020-08-21 at 22 20 11
Screenshot 2020-08-21 at 20 59 08

@bigcommerce/checkout @bigcommerce/payments

### CODE REVIEW

**Comment by bc-maxym-parfenov** [comment]
@bigcommerce/checkout

File: docs/interfaces/steptracker.md [hunk 1]
```diff
@@ -27,7 +27,13 @@ ___
 
 ###  trackOrderComplete
 
-▸ **trackOrderComplete**(): *void*
+▸ **trackOrderComplete**(`features?`: Record‹string, boolean›): *void*
```
Comment (capsula4) [inline docs/interfaces/steptracker.md:30]: you don't need to commit these, `docs` is autogenerated.

File: src/analytics/analytics-step-tracker.ts [hunk 2]
```diff
@@ -99,7 +99,7 @@ export default class AnalyticsStepTracker implements StepTracker {
         this._checkoutStarted = true;
     }
 
-    trackOrderComplete(): void {
+    trackOrderComplete(features?: Record<string, boolean>): void {
```
Comment (capsula4) [inline src/analytics/analytics-step-tracker.ts:102]: might be worth adding a docblock explaining how `features work`.

The docblock will automatically be included in the generated documentation

File: src/analytics/analytics-step-tracker.ts [hunk 3]
```diff
@@ -137,7 +137,34 @@ export default class AnalyticsStepTracker implements StepTracker {
...
+                '&ta': payload.affiliation,
+                '&tr': payload.revenue,
+                '&ts': payload.shipping,
+                '&tt': payload.tax,
+                '&tcc': payload.coupon,
+                '&cu': payload.currency,
+            });
+
+            payload.products.map(product => {
+                this.analytics.hit('item', {
```
Comment (capsula4) [inline src/analytics/analytics-step-tracker.ts:154]: this object seems to be the same as the previous one, might be worth re-using it?

```js
const compressedPayload = {
                '&ti': payload.orderId,
                '&ta': payload.affiliation,
                '&tr': payload.revenue,
                '&ts': payload.shipping,
                '&tt': payload.tax,
                '&tcc': payload.coupon,
                '&cu': payload.currency,
            })

**Review by capsula4 — COMMENTED** [review]
just one minor comment. Would be good to maybe add a unit test?

**Comment by bc-maxym-parfenov** [comment]
> just one minor comment. Would be good to maybe add a unit test?
Unit test for experiment case added.

**Review by capsula4 — APPROVED** [review]
:shipit:

File: src/analytics/analytics-step-tracker.ts [hunk 4]
```diff
@@ -137,7 +140,34 @@ export default class AnalyticsStepTracker implements StepTracker {
...
+                '&ti': payload.orderId,
+                '&ta': payload.affiliation,
+                '&tr': payload.revenue,
+                '&ts': payload.shipping,
+                '&tt': payload.tax,
+                '&tcc': payload.coupon,
+                '&cu': payload.currency,
+            });
+
+            payload.products.map(product => {
```
Comment (icatalina) [inline src/analytics/analytics-step-tracker.ts:156]: This should have used `forEach` since there's no return value.

**Pull request merged**